quickwit/quickwit-search/src/top_k_collector.rs

```diff
@@ -0,0 +1,860 @@
+            SpecializedSegmentTopKCollector::<ReverseOptionU64, (), true>::new(
+                split_id,
+                score_extractor,
+                leaf_max_hits,
+                segment_ord,
+            ),
+        ),
+        (SortType::OneFFSort, SortOrder::Desc, SortOrder::Asc) => Box::new(
+            SpecializedSegmentTopKCollector::<Option<u64>, (), false>::new(
+                split_id,
+                score_extractor,
+                leaf_max_hits,
+                segment_ord,
+            ),
+        ),
+        (SortType::OneFFSort, SortOrder::Asc, SortOrder::Desc) => Box::new(
+            SpecializedSegmentTopKCollector::<ReverseOptionU64, (), true>::new(
+                split_id,
+                score_extractor,
+                leaf_max_hits,
+                segment_ord,
+            ),
+        ),
+        (SortType::OneFFSort, SortOrder::Desc, SortOrder::Desc) => Box::new(
+            SpecializedSegmentTopKCollector::<Option<u64>, (), false>::new(
+                split_id,
+                score_extractor,
+                leaf_max_hits,
+                segment_ord,
+            ),
+        ),
+        (SortType::TwoFFSorts, SortOrder::Asc, SortOrder::Asc) => {
+            Box::new(SpecializedSegmentTopKCollector::<
+                ReverseOptionU64,
+                ReverseOptionU64,
+                true,
+            >::new(
+                split_id, score_extractor, leaf_max_hits, segment_ord
+            ))
+        }
+        (SortType::TwoFFSorts, SortOrder::Asc, SortOrder::Desc) => {
+            Box::new(SpecializedSegmentTopKCollector::<
+                ReverseOptionU64,
+                Option<u64>,
+                true,
+            >::new(
+                split_id, score_extractor, leaf_max_hits, segment_ord
+            ))
+        }
+        (SortType::TwoFFSorts, SortOrder::Desc, SortOrder::Asc) => {
+            Box::new(SpecializedSegmentTopKCollector::<
+                Option<u64>,
+                ReverseOptionU64,
+                false,
+            >::new(
+                split_id, score_extractor, leaf_max_hits, segment_ord
+            ))
+        }
+        (SortType::TwoFFSorts, SortOrder::Desc, SortOrder::Desc) => {
+            Box::new(SpecializedSegmentTopKCollector::<
+                Option<u64>,
+                Option<u64>,
+                false,
+            >::new(
+                split_id, score_extractor, leaf_max_hits, segment_ord
+            ))
+        }
+    }
+}
+
+/// Fast Top K Computation
+///
+/// The buffer is truncated to the top_n elements when it reaches the capacity of the Vec.
+/// That means capacity has special meaning and should be carried over when cloning or serializing.
+///
+/// For TopK == 0, it will be relative expensive.
+pub struct TopKComputer<D> {
+    /// Reverses sort order to get top-semantics instead of bottom-semantics
+    buffer: Vec<Reverse<D>>,
+    top_n: usize,
+    pub(crate) threshold: Option<D>,
```

Comment:
Could we remove the Option in threshold?
(and initialize it to Max or Min depending on what we do)

```suggestion
    pub(crate) threshold: Option<D>,
```

Reply:
Yes, seems to be slightly faster